src/lib/y2network/clients/routing.rb

```diff
@@ -0,0 +1,494 @@
+      end
+
+      # Handler for action "add"
+      #
+      # @param options [Hash{String => String}] action options
+      # @return [Boolean] true if added; false otherwise
+      def AddHandler(options)
+        destination = options["dest"]
+
+        unless destination
+          CommandLine.Error(_("At least destination must be specified."))
+          return false
+        end
+
+        route = route_from_options(options)
+
+        CommandLine.Print(
+          Builtins.sformat(_("Adding '%1' destination to routing table ..."), destination)
+        )
+
+        yast_routing_table.routes << route if !yast_routing_table.routes.include?(route)
+
+        true
+      end
+
+      # Handler for action "edit"
+      #
+      # @param options [Hash{String => String}] action options
+      # @return [Boolean] true if edited; false otherwise
+      def EditHandler(options)
+        destination = options["dest"]
+
+        if destination == ""
+          CommandLine.Error(_("Destination IP address must be specified."))
+          return false
+        end
+
+        if options.size < 2
+          CommandLine.Error(
+            _(
+              "At least one of the following parameters (gateway, netmask, device, options) must be specified"
+            )
+          )
+          return false
+        end
+
+        route = destination_routes(destination).first
+
+        if route
+          edited_route = route_from_options(options)
+          CommandLine.Print(
+            Builtins.sformat(
+              _("Updating '%1' destination in routing table ..."),
+              destination
+            )
+          )
+
+          update_route_with!(route, edited_route)
+        else
+          CommandLine.Error(
+            Builtins.sformat(
+              _("No entry for destination '%1' in routing table"),
+              destination
+            )
+          )
+          return false
+        end
+
+        true
+      end
+
+      # Handler for action "delete"
+      #
+      # @param [Hash]
+      # @return [Boolean] true if deleted; false otherwise
+      def DeleteHandler(options)
+        destination = options["dest"]
+        route = destination_routes(destination).first
+
+        if route
+          CommandLine.Print(
+            Builtins.sformat(
+              _("Deleting '%1' destination from routing table ..."),
+              destination
+            )
+          )
+          yast_routing_table.routes.delete(route)
+        else
+          CommandLine.Error(
+            Builtins.sformat(
+              _("No entry for destination '%1' in routing table"),
+              destination
+            )
+          )
+        end
+
+        !!route
+      end
+
+      # Initialize the yast and system {Y2Network::Config}s from the sysconfig
+      #
+      # @return [Boolean]
+      def read
+        Yast::Lan.add_config(:yast, Y2Network::Config.from(:sysconfig))
```

Comment:
hmm, this basically means we are reading it twice. What about just copying data there?